### PR TITLE
MINOR: adding filter variable for dev/tasks

### PR DESCRIPTION
**Pull request by sunnysideup**
Long overdue change to find tasks easily. 

Tasks can be filtered by adding the `q` request variable (e.g. `/dev/tasks/?q=test` OR `vendor/bin/sake dev/tasks q=test`)

### CODE REVIEW

**Comment by sunnysideup** [comment]
_I'm assuming this is mostly for use in the browser, since getting tasks with sake you can just pipe into grep for the same functionality._ 

Not from my point of view. This requires you to know how to use grep, etc... I would keep it simpler (or provide some information on how to use grep).

File: src/Dev/TaskRunner.php [hunk 1]
```diff
@@ -42,9 +42,28 @@ protected function init()
...
+     * list of options - can be filtered by adding the `q` request variable (e.g. `/dev/tasks/?q=test` OR `vendor/bin/sake dev/tasks q=test`)
+     * @param  HTTPRequest $index
+     * @return DBHTMLText|string - depends on whether it is `Director::is_cli()` request (returns string) or not (returns DBHTMLText)
+     */
+    public function index($request = null)
     {
+        $baseUrl = Director::absoluteBaseURL();
         $tasks = $this->getTasks();
+        $filter = (string) trim($request->requestVar('q'));
+        if($filter) {
```
[the remark this answers is not in the document]
Comment (sunnysideup) [reply src/Dev/TaskRunner.php]: Fixed

File: src/Dev/TaskRunner.php [hunk 2]
```diff
@@ -42,9 +42,28 @@ protected function init()
...
+    /**
+     * list of options - can be filtered by adding the `q` request variable (e.g. `/dev/tasks/?q=test` OR `vendor/bin/sake dev/tasks q=test`)
+     * @param  HTTPRequest $index
+     * @return DBHTMLText|string - depends on whether it is `Director::is_cli()` request (returns string) or not (returns DBHTMLText)
+     */
+    public function index($request = null)
     {
+        $baseUrl = Director::absoluteBaseURL();
         $tasks = $this->getTasks();
+        $filter = (string) trim($request->requestVar('q'));
```
Comment (michalkleiner) [inline src/Dev/TaskRunner.php:54]: `trim` already casts to string, so the extra casting is not necessary

Comment (sunnysideup) [reply src/Dev/TaskRunner.php:54]: true!

**Comment by sunnysideup** [comment]
If you guys like it then I will make a new pull request with everything sorted. I am not sure about the form though as we dont want to make it too fancy. Just simple.

**Comment by sunnysideup** [comment]
Maybe we can add a little form, but then I think this should be Javascript based, not PHP based.

File: src/Dev/TaskRunner.php [hunk 3]
```diff
@@ -42,9 +42,28 @@ protected function init()
...
 
-    public function index()
+    /**
+     * list of options - can be filtered by adding the `q` request variable (e.g. `/dev/tasks/?q=test` OR `vendor/bin/sake dev/tasks q=test`)
+     * @param  HTTPRequest $index
+     * @return DBHTMLText|string - depends on whether it is `Director::is_cli()` request (returns string) or not (returns DBHTMLText)
+     */
+    public function index($request = null)
     {
+        $baseUrl = Director::absoluteBaseURL();
```
Comment (michalkleiner) [inline src/Dev/TaskRunner.php:52]: This line does nothing here by the looks, is it needed?

**Comment by michalkleiner** [comment]
My 2c — even though it's quite a small addition, I personally think it's not a scope we want to support within the framework in the long term, mainly because there are other quite standard ways how to achieve the same without this — as Guy mentioned, in the CLI folks can use grep or other filtering mechanisms, in the browser you can search within the content of the page using CTLR/CMD+F shortcut.

**Comment by sunnysideup** [comment]
Yeah - then lets leave it. Dont you think? I guess what would be awesome if we had some sort of autocomplete. I usually use CTRL+F on the screen, but most of the time I run it from command line and then I just copy the output and do a CTRL+F.

Maybe it is better to have the tasks grouped or something like that as we work on lots of projects with 50 or so tasks - where it is a real pain to find the right one.

**Comment by GuySartorelli** [comment]
> Yeah - then lets leave it.
Closing this PR

**Pull request closed without merging**